packages/react-lowcode/src/codegen/generation/generators/list/mui-datatable-generator.ts

```diff
@@ -251,6 +255,46 @@ export default class MuiDataTableGenerator implements TableGenerator
       }else return undefined
     }
 
+    generateTablePage(template: string): string | undefined {
+      let result: string | undefined;
+
+      if (this._entity) {
+        let ast = createAst(template);
+
+        if (ast) {
+          const templateParenthesizedExpression = find<ParenthesizedExpression>(ast, (node: Node) => {
+            return ts.isParenthesizedExpression(node);
+          });
+
+          if (templateParenthesizedExpression) {
+            const tableComponentName = this._helper.getComponentName(this._entity);
+            const inputParameterIdentifier = this._helper.getInputParameterIdentifier(this._entity);
+
+            const parenthesizedExpression = factory.createParenthesizedExpression(factory.createJsxSelfClosingElement(
+              factory.createIdentifier(tableComponentName),
+              undefined,
+              factory.createJsxAttributes([factory.createJsxAttribute(
+                inputParameterIdentifier,
+                factory.createJsxExpression(
+                  undefined,
+                  factory.createPropertyAccessChain(
+                    factory.createIdentifier("data"),
```

Comment:
The prop `data` should not be hardcoded here (at least const for default, ideally try to propagate use the prop from table  component)